### PR TITLE
ARROW-3376: [C++] Add double-conversion to cpp/thirdparty/download_dependencies.sh

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -169,10 +169,10 @@ else()
   set(BROTLI_SOURCE_URL "https://github.com/google/brotli/archive/${BROTLI_VERSION}.tar.gz")
 endif()
 
-if (DEFINED ENV{DOUBLE_CONVERSION_SOURCE_URL})
-  set(DOUBLE_CONVERSION_SOURCE_URL "$ENV{DOUBLE_CONVERSION_SOURCE_URL}")
+if (DEFINED ENV{ARROW_DOUBLE_CONVERSION_URL})
```
Comment (kszucs) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake]: Renamed to follow `ARROW_{dependency}` naming convention.

Comment (pitrou) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Hmm, right, but please keep `DOUBLE_CONVERSION_SOURCE_URL` for the internal CMake variable (as in `BROTLI_SOURCE_URL` above).

**Review by pitrou — COMMENTED** [review]
Fine on the principle, just one comment.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (kszucs) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake]: Oops, fixed it!

**Review by wesm — APPROVED** [review]
+1

**Pull request closed without merging**